packages/image_picker/image_picker/ios/Classes/FLTPHPickerSaveImageToPathOperation.m

```diff
@@ -82,51 +84,63 @@ - (void)start {
   }
   if (@available(iOS 14, *)) {
     [self setExecuting:YES];
+
+    if ([self.result.itemProvider hasItemConformingToTypeIdentifier:UTTypeWebP.identifier]) {
+      [self.result.itemProvider
+          loadDataRepresentationForTypeIdentifier:UTTypeWebP.identifier
+                                completionHandler:^(NSData *_Nullable data,
+                                                    NSError *_Nullable error) {
+                                  UIImage *image = [[UIImage alloc] initWithData:data];
```

Comment:
Can we also add error handling here? What if `data` is nil? (Should also test this)

Reply:
You are right, we have to add error handling here. The reason why I didn't do it because I thought it's outside of my scope :D But I will add it anyway.

Reply:
@cyanglaz 

I think it's not easy to add error handling according the code that we have. Let see the signature of the function in flutter:
```
Future<List<PickedFile>?> pickMultiImage({
    double? maxWidth,
    double? maxHeight,
    int? imageQuality,
  })
```

It's not clear how developer can handle the error because there are many scenarios:
1. All selected images return error.
2. Many of selected images return error.

In the first case we throw just error. But in second case throwing the error doesn't make sense, because I think the devs still want to get successfully picked image, and then show the error for those images, that throw error. 

Passing a successfully picked image through error can solve the problem, but I think it's not a good design.

My suggestion is creating something like that:
```
class PickedFileInfo {
   final PickedFile file;
   final Error? error;
}

Future<List<PickedFileInfo>?> pickMultiImage({
    double? maxWidth,
    double? maxHeight,
    int? imageQuality,
  })
```

or

```
Future<List<PickedFileInfo>?> pickMultiImage({
    double? maxWidth,
    double? maxHeight,
    int? imageQuality,
    void function(error)? onError,
  })
```

Reply:
This is a good point. Can we instead add a new field called `error` in PickeFile?

Reply:
Ok, I will try. I let you know later.

Reply:
Since this PR already improves the current plugin and error handling is out of scope, I'm ok with not adding it in this PR. I create an issue tracking the error handling here: https://github.com/flutter/flutter/issues/95268